linalg/vector.hpp

```diff
@@ -36,6 +36,10 @@
 namespace mfem
 {
 
+// Optional MemoryType for the reduction buffer when using CUDA or HIP,
+// the default is MemoryType::DEFAULT
+extern mfem::MemoryType device_reduce_buf_type;
```

Comment:
Using pinned host memory may be the best option for this in most cases -- should we make that the default?

Reply:
Yes, a good idea!